tests/migration/openldap_to_389ds.pm

```diff
@@ -0,0 +1,95 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Implement opendlap to 389ds migration tools test case into openQA
+#
+# Maintainer: wegao <wegao@suse.com>
+
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use Utils::Systemd qw(systemctl disable_and_stop_service);
+use registration qw(add_suseconnect_product);
+
+sub run {
+    select_console("root-console");
+
+    # Install 389-ds and sssd on client
+    zypper_call("in 389-ds sssd sssd-tools");
+    script_run "rpm -qa 389-ds";
+    # Install openldap since we need use slaptest tools
+    add_suseconnect_product('sle-module-legacy');
+    zypper_call("in openldap2 sssd-ldap sssd-tools openldap2-client");
+
+    # Disable and stop the nscd daemon because it conflicts with sssd
+    disable_and_stop_service("nscd", ignore_failure => 1);
+
+    # Configure nsswitch with sssd
+    script_run "sed -i 's/^passwd:.*/passwd: compat sss/' /etc/nsswitch.conf";
+    script_run "sed -i 's/^group:.*/group: compat sss/' /etc/nsswitch.conf";
+    script_run "cat /etc/nsswitch.conf";
+
+    # Prepare test env
+    script_run "cd; curl -L -v " . autoinst_url . "/data/openldap_to_389ds > openldap_to_389ds.data && cpio -id < openldap_to_389ds.data && mv data test && ls test";
+    script_run("cd test");
+
+    # We need start openldap to kick out date base file which stored in directory
+    script_run "mkdir /tmp/ldap-sssdtest";
+    script_run "slapd -h 'ldap:///' -f slapd.conf";
+    script_run "ldapadd -x -D 'cn=root,dc=ldapdom,dc=net' -wpass -f db.ldif";
+    script_run "killall slapd";
+    script_run "ps -aux | grep slapd";
+
+    # setup sssd
+    script_run "cp ./sssd.conf /etc/sssd/sssd.conf";
+    systemctl("restart sssd", ignore_failure => 1);
+    systemctl("status sssd",  ignore_failure => 1);
+    script_run "sss_cache -E";
+
+    # Prepare data file for migration
+    script_run "mkdir slapd.d";
+    script_run("dscreate from-file ./instance.inf", timeout => 60);
+    script_run "dsctl localhost status";
+    script_run "slaptest -f slapd.conf -F ./slapd.d";
+
+    # Check migration tools
+    script_run "openldap_to_ds --confirm localhost ./slapd.d ./db.ldif";
+    script_run "ldapmodify -H ldap://localhost -x -D 'cn=Directory Manager' -w YOUR_ADMIN_PASSWORD_HERE -f aci.ldif";
+
+    # Manual fix memberof plugin
+    script_run "dsconf localhost plugin memberof show";
+    script_run "systemctl restart dirsrv\@localhost";
+    script_run "dsconf localhost plugin memberof fixup dc=ldapdom,dc=net -f '(objectClass=*)'";
+
+    # check memeberof plugin
+    validate_script_output("ldapsearch -H ldap://localhost -b 'dc=ldapdom,dc=net' -s sub -x -D 'cn=Directory Manager' -w YOUR_ADMIN_PASSWORD_HERE memberof", sub { m/memberof:.*group1/ });
+    # wait a second before check getent
+    sleep 5;
```

Comment:
Is any reason add wait time here?

Reply:
That's trick part, sometimes the getent seems not stable, so i put some sleep here.

Reply:
If that's the reason, express it better in the comment:

```
Wait an arbitrary ammount of time to give getent time be able to locate the users in the database
```

Reply:
Issue "sync" command may help there, in cases the database is not updated in short time. we can have a try :)

Reply:
I will create issue for this later, remove this line